### PR TITLE
Matching log-source tagging for diagnosability with previous apt changes

### PR DESCRIPTION
**Pull request by kjohn-msft**
This PR makes no functional changes to the code beyond:
- more clarity in logging origin through tagging with source file acronyms. 
- deprioritizing some logs (debug to verbose) that are in years-old highly stable code paths.

This is similar to code that's already gone in for Apt.

### CODE REVIEW

File: src/core/src/package_managers/YumPackageManager.py [hunk 1]
```diff
@@ -291,21 +285,21 @@ def is_package_version_installed(self, package_name, package_version):
         # Loaded plugins: product-id, search-disabled-repos, subscription-manager
         # Installed Packages
         # kernel.x86_64                                                                                   3.10.0-514.el7                                                                                    @anaconda/7.3
-        self.composite_logger.log_debug("\nCHECKING PACKAGE INSTALL STATUS FOR: " + str(package_name) + " (" + str(package_version) + ")")
+        self.composite_logger.log_verbose("[YPM] Checking package install status. [PackageName={0}][PackageVersion={1}]".format(package_name, package_version))
```
Comment (rane-rajasi) [inline src/core/src/package_managers/YumPackageManager.py:288]: nit: ___.format(str(package_name), str(package_version))

File: src/core/src/package_managers/YumPackageManager.py [hunk 2]
```diff
@@ -291,21 +285,21 @@ def is_package_version_installed(self, package_name, package_version):
...
+                self.composite_logger.log_debug("[YPM] > Installed version match found. [PackageName={0}][PackageVersion={1}]".format(package_name, package_version))
                 return True
             else:
-                self.composite_logger.log_debug(" - Did not match: " + package + " (" + package_versions[index] + ")")
+                self.composite_logger.log_verbose("[YPM] > Did not match: " + package + " (" + package_versions[index] + ")")
 
         # sometimes packages are removed entirely from the system during installation of other packages
         # so let's check that the package is still needed before
-
+        self.composite_logger.log_debug(" - Installed version match NOT found for: " + str(package_name) + "(" + str(package_version) + ")")
```
Comment (rane-rajasi) [inline src/core/src/package_managers/YumPackageManager.py:302]: nit: self.composite_logger.log_debug(" - Installed version match NOT found for: [PackageName={0}][PackageVersion={1}]".format(str(package_name), str(package_version)))

File: src/core/src/package_managers/YumPackageManager.py [hunk 3]
```diff
@@ -863,42 +853,36 @@ def try_mitigate_issues_if_any(self, command, code, out):
...
-        self.composite_logger.log_debug("\nUpdating client package to avoid errors from older certificates using command: [Command={0}]".format(str(command)))
+        self.composite_logger.log_debug("[Customer-environment-error] Updating client package to avoid errors from older certificates using command: [Command={0}]".format(str(command)))
         code, out = self.env_layer.run_command_output(command, False, False)
         if code != self.yum_exitcode_no_applicable_packages:
-            self.composite_logger.log('[ERROR] Package manager was invoked using: ' + command)
-            self.composite_logger.log_warning(" - Return code from package manager: " + str(code))
-            self.composite_logger.log_warning(" - Output from package manager: \n|\t" + "\n|\t".join(out.splitlines()))
-            self.telemetry_writer.write_execution_error(command, code, out)
-            error_msg = 'Unexpected return code (' + str(code) + ') from package manager on command: ' + command
+            error_msg = 'Customer environment error (expired SSL certs):  [Command={0}][Code={1}][Out={2}]'.format(command,str(code),out)
```
Comment (rane-rajasi) [inline src/core/src/package_managers/YumPackageManager.py:878]: nit: we should have str(out) in format(...) incase out has any special characters

File: src/core/src/package_managers/YumPackageManager.py (hunk 1 above)
Comment (kjohn-msft) [reply src/core/src/package_managers/YumPackageManager.py]: Addressed.

File: src/core/src/package_managers/YumPackageManager.py (hunk 2 above)
Comment (kjohn-msft) [reply src/core/src/package_managers/YumPackageManager.py]: Addressed.

File: src/core/src/package_managers/YumPackageManager.py (hunk 3 above)
Comment (kjohn-msft) [reply src/core/src/package_managers/YumPackageManager.py]: I realized we don't want this addition to go into status, only into the logs.
Moved out into just the error logging.
str(x) is only required if there's a risk the type of input is not str. Many places here unnecessarily have it but it's forcing type safety (good thing).

**Pull request merged**